src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -2,6 +2,7 @@
 import logging
 
 from pip._vendor import six
+from pip._vendor.packaging.specifiers import SpecifierSet
```

Comment:
I think this needs to be moved back into the `if MYPY_CHECK_RUNNING` block (I wish flake8 is smarter and can detect this unused import automatically)

Reply:
Good catch, ta. Updated

Reply:
Also, I wish typing didn't need that runtime `if MYPY_CHECK_RUNNING` approach that makes flake8 unable to detect the unused import. But that's just another way of saying the same thing, basically.